app/ide-desktop/lib/client/src/config/parser.ts

```diff
@@ -316,35 +311,38 @@ export function parseArgs() {
     interface YargsArgs {
         // We don't control the naming of this third-party API.
         /* eslint-disable @typescript-eslint/naming-convention */
-        [key: string]: string[] | string | undefined
+        [key: string]: string[] | string
         _: string[]
         // Exists only when the `populate--` option is enabled.
         '--'?: string[]
         $0: string
         /* eslint-enable @typescript-eslint/naming-convention */
     }
 
-    let parseError: Error | undefined
+    // Required otherwise TypeScript thinks it's always `null`.
+    // eslint-disable-next-line no-restricted-syntax
+    let parseError = null as Error | null
+    // Required since `parse` is defined to potentially return a `Promise`.
+    // This only happens when an async middleware has been registered,
+    // which we are not doing.
+    // eslint-disable-next-line no-restricted-syntax
     const { '--': unexpectedArgs, ...parsedArgs } = optParser.parse(
         argv,
         {},
         // @ts-expect-error Yargs' typings are wrong.
-        // eslint-disable-next-line no-restricted-syntax
         (err: Error | null) => {
-            if (err) {
+            if (err != null) {
                 parseError = err
             }
         }
     ) as YargsArgs
```

Comment:
type assertion #5

Reply:
if you see any way of fixing some of the type casts, lets do it. If not, lets not spend too much time on it for now :)